testing/citest/tests/azure_bake_and_deploy_test.py

```diff
@@ -0,0 +1,715 @@
+            "addressPrefix": self.__subnet[1]['ADDRESS'],
+            "device": [],
+            "id": '/subscriptions/{id}/resourceGroups/{rg}/providers/Microsoft.Network/virtualNetworks/{vnet}/subnets/{name}'.format(
+                id=self.__subscription_id, rg=self.__rg_name,
+                vnet=self.__vnet_name, name=self.__subnet[1]['NAME']
+            ),
+            "name": self.__subnet[1]['NAME'],
+            "purpose": 'TBD',
+            "region": self.__rg_location,
+            "type": 'azure',
+            "vnet": self.__vnet_name
+        }]
+        vnet = {
+            "account": self.ACCOUNT,
+            "cloudProvider": "azure",
+            "id": self.__vnet_name,
+            "name": self.__vnet_name,
+            "region": self.__rg_location,
+            "resourceGroup": self.__rg_name,
+            "subnets": subnets
+        }
+        payload = self.agent.make_json_payload_from_kwargs(
+            job=[{
+                "stack": self.__stack,
+                "detail": self.__detail,
+                "credentials": self.ACCOUNT,
+                "region": self.__rg_location,
+                "cloudProvider": "azure",
+                "vnet": self.__vnet_name,
+                "subnet": self.__subnet[1]['NAME'],
+                "probes": healthyCheck,
+                "securityGroups": [],
+                "loadBalancingRules": rules,
+                "name": self.__full_lb_name,
+                "selectedVnet": vnet,
+                "vnetResourceGroup": self.__rg_name,
+                "selectedSubnet": subnets[0],
+                "type": "upsertLoadBalancer",
+                "loadBalancerType": "Azure Application Gateway",
+                "appName": self.TEST_APP,
+                "loadBalancerName": self.__full_lb_name,
+                "user": "[anonymous]"
+            }],
+            description="Test - Create load balancer: {lb}".format(
+                lb=self.__full_lb_name),
+            application=self.TEST_APP)
+
+        builder = az.AzContractBuilder(self.az_observer)
+        (builder.new_clause_builder(
+            'Load Balancer Created', retryable_for_secs=30)
+        .collect_resources(
+            az_resource='network',
+            command='application-gateway',
+            args=['list', '--resource-group',
+            '{app}-{rg}'.format(app=self.TEST_APP, rg=self.__rg_location)])
+        .EXPECT(ov_factory.value_list_contains(
+            jp.DICT_MATCHES({
+                'name': jp.STR_EQ(self.__full_lb_name),
+                'tags': jp.DICT_MATCHES({
+                    'vnet': jp.STR_EQ('demo1-vn-zhijie'),
+                    'subnet': jp.STR_EQ('subnet2')
+                })
+            }))))
+
+        return st.OperationContract(
+            self.new_post_operation(
+                title="create_load_balancer", data=payload,
+                path=('applications/{app}/tasks').format(app=self.TEST_APP),
+                max_wait_secs=2400),
```

Comment:
> Is this intentionally 40 minutes, or did you mean for it to be only 4?

Provisioning an Azure Application Gateway that plays the role of the Spinnaker Load Balancer requires at least 15 minutes, which is by Azure design. Setting 40 min here since we did run into the bad situation when doing offline testing.